picard/formats/id3.py

```diff
@@ -1035,6 +1186,49 @@ def _remove_other_supported_tag(self, tags, real_name):
         """Remove other supported tag from ID3 frames."""
         del tags[real_name]
 
+    def _process_link_frames_on_load(self, tags, metadata):
+        """Process malformed LINK frames during initial file load and add them as website URLs."""
+
+        link_frames = []
+        for frame_id, frame in list(tags.items()):
+            if frame_id.startswith('LINK'):
+                is_potentially_malformed = False
+
+                if ':' in frame_id and any(term in frame_id for term in ('http', 'www.', '://')):
+                    is_potentially_malformed = True
+
+                if hasattr(frame, 'data') and frame.data:
+                    try:
+                        frame_data = frame.data.decode('latin1', errors='ignore')
+
+                        if frame_data.startswith('\x00') and any(term in frame_data[1:] for term in ('http', 'www.', '://')):
+                            is_potentially_malformed = True
+
+                        elif '\x00' in frame_data:
+                            parts = frame_data.split('\x00', 1)
+                            if len(parts) > 1 and any(term in parts[1] for term in ('http', 'www.', '://')):
+                                is_potentially_malformed = True
+
+                        elif any(term in frame_data for term in ('http', 'www.', '://')) and '\x00' not in frame_data:
+                            is_potentially_malformed = True
+                    except Exception:
+                        is_potentially_malformed = True
+
+                if is_potentially_malformed:
+                    link_frames.append((frame_id, frame))
+        
+        if not link_frames:
+            return
+
+        to_add = []
+        for frame_id, frame in link_frames:
+            if self._extract_and_convert_link_frame(frame, frame_id, to_add):
```

Comment:
`_extract_and_convert_link_frame()` could be made a generator (using `yield`), instead of passing `to_add`.